api/commands/test.ts

```diff
@@ -14,10 +14,20 @@ async function runTest(
 ): Promise<dataform.ITestResult> {
   // TODO: Test results are currently limited to 1000 rows.
   // We should paginate test results to remove this limit.
-  const [actualResults, expectedResults] = await Promise.all([
-    dbadapter.execute(testCase.testQuery, { maxResults: 1000 }),
-    dbadapter.execute(testCase.expectedOutputQuery, { maxResults: 1000 })
-  ]);
+  let actualResults;
+  let expectedResults;
+  try {
+    [actualResults, expectedResults] = await Promise.all([
+      dbadapter.execute(testCase.testQuery, { maxResults: 1000 }),
+      dbadapter.execute(testCase.expectedOutputQuery, { maxResults: 1000 })
+    ]);
+  } catch (e) {
+    return {
+      name: testCase.name,
+      successful: false,
+      messages: [`Error thrown: ${e.message}.`]
```

Comment:
stack?

Reply:
In this case it *should* only really ever be errors coming straight from the DB, so I'm going with `message` to keep it short. But if we notice it's hiding stuff, we can definitely change this